src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,146 @@
+import com.google.api.codegen.transformer.PageStreamingTransformer;
+import com.google.api.codegen.transformer.PathTemplateTransformer;
+import com.google.api.codegen.transformer.ServiceTransformer;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.util.ruby.RubyTypeTable;
+import com.google.api.codegen.viewmodel.ApiMethodView;
+import com.google.api.codegen.viewmodel.DynamicLangXApiView;
+import com.google.api.codegen.viewmodel.GrpcStreamingDetailView;
+import com.google.api.codegen.viewmodel.LongRunningOperationDetailView;
+import com.google.api.codegen.viewmodel.PathTemplateGetterFunctionView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+
+/** The ModelToViewTransformer to transform a Model into the standard GAPIC surface in Ruby. */
+public class RubyGapicSurfaceTransformer implements ModelToViewTransformer {
+  private static final String XAPI_TEMPLATE_FILENAME = "ruby/main.snip";
+
+  private final FileHeaderTransformer fileHeaderTransformer =
+      new FileHeaderTransformer(new RubyImportSectionTransformer());
+  private final DynamicLangApiMethodTransformer apiMethodTransformer =
+      new DynamicLangApiMethodTransformer(new RubyParamDocTransformer());
+  private final ServiceTransformer serviceTransformer = new ServiceTransformer();
+  private final GrpcStubTransformer grpcStubTransformer = new GrpcStubTransformer();
+  private final PageStreamingTransformer pageStreamingTransformer = new PageStreamingTransformer();
+  private final BundlingTransformer bundlingTransformer = new BundlingTransformer();
+  private final PathTemplateTransformer pathTemplateTransformer = new PathTemplateTransformer();
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return ImmutableList.of(XAPI_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      ModelTypeTable modelTypeTable =
+          new ModelTypeTable(
+              new RubyTypeTable(apiConfig.getPackageName()),
+              new RubyModelTypeNameConverter(apiConfig.getPackageName()));
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(
+              service,
+              apiConfig,
+              modelTypeTable,
+              new RubySurfaceNamer(apiConfig.getPackageName()),
+              new RubyFeatureConfig());
+      surfaceDocs.add(createView(context));
+    }
+    return surfaceDocs;
+  }
+
+  private ViewModel createView(SurfaceTransformerContext context) {
```

Comment:
I have a hunch that this can be abstracted out. Lets not do that for this PR, but it would be good for us to keep that in mind for the next two conversions.